update_keb_installation.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# workflow-helpers
+# Copyright(C) 2021 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Update Kebechet installation details in DB during webhook workflows."""
+
+import os
+from contextlib import contextmanager
+from tempfile import TemporaryDirectory
+
+import git
+import yaml
+import json
+
+from thoth.storages import GraphDatabase
+from thamos.cli import _load_files
+from thoth.common import logging
+
+from thoth.workflow_helpers import __service_version__
+
+_SLUG = json.loads(os.getenv("GITHUB_SLUG") or "null")
+_ACCESS_TOKEN = os.getenv("KEBECHET_ACCESS_TOKEN")
+
+_APP_NAME = "KEBECHET"
+_GITHUB_URL = "https://github.com"
+
+db = GraphDatabase()
+db.connect()
+
+_LOGGER = logging.getLogger("thoth.update_kebechet_installation")
+_LOGGER.info("Thoth workflow-helpers task: update kebechet installation details v%s", __service_version__)
+
+
+#########  Copied from https://github.com/thoth-station/kebechet in kebechet/utils.py #############
+@contextmanager
+def cwd(path: str):
+    """Change working directory in a push-pop manner with context manager."""
+    previous_dir = os.getcwd()
+    try:
+        os.chdir(path)
+        yield previous_dir
+    finally:
+        os.chdir(previous_dir)
+
+
+@contextmanager
+def cloned_repo(**clone_kwargs):
+    """Clone the given Git repository and cd into it."""
+    repo_url = f"https://{_APP_NAME}:{_ACCESS_TOKEN}@{_GITHUB_URL}/{_SLUG}"
+
+    with TemporaryDirectory() as repo_path, cwd(repo_path):
+        _LOGGER.info(f"Cloning repository {repo_url} to {repo_path}")
+        repo = git.Repo.clone_from(repo_url, repo_path, **clone_kwargs)
```

Comment:
This might be very expensive operation, based on the repo size. As you need just `.thoth.yaml`, can it be directly obtained by checking raw files? E.g. for this repo it is:

https://raw.githubusercontent.com/thoth-station/workflow-helpers/master/.thoth.yaml

The URL can be constructed and there is no request throttling AFAIK.

Reply:
@fridex This is the approach that I used in Kebechet, but it is causing issues now because I had to make assumptions regarding the default branch. For instance it does not work if default branch is `main`. If we have some way of getting the default branch then we could do this easily.

Should be possible using the `GitHub` api. https://bit.ly/3rgSaDx

Reply:
> @fridex This is the approach that I used in Kebechet, but it is causing issues now because I had to make assumptions regarding the default branch. For instance it does not work if default branch is `main`. If we have some way of getting the default branch then we could do this easily.
> 
> Should be possible using the `GitHub` api. https://bit.ly/3rgSaDx

+1, I think we wanted to have this logic in source-management in kebechet for updates (cannot find related issue). That will require GH auth setup and stuff like that. Maybe short-term cloning (with `depth=1`) could be a good (short-term) approach then. :+1: